### PR TITLE
PP-7595 Switch payment links controllers to use asnyc/await

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Replaces promise code to use asnyc/await syntax which makes the code easier to follow.

### CODE REVIEW

**Review by oswaldquek — APPROVED** [review]
LGTM

File: app/controllers/payment-links/get-delete.controller.js [hunk 1]
```diff
@@ -6,30 +6,27 @@ const productsClient = require('../../services/clients/products.client.js')
...
+
+    const deleteToken = publicAuthClient.deleteTokenForAccount({
+      accountId: gatewayAccountId,
+      correlationId: req.correlationId,
+      payload: {
+        token: product.apiToken
+      }
     })
+    const deleteProduct = productsClient.product.delete(gatewayAccountId, req.params.productExternalId)
+    await Promise.all([deleteToken, deleteProduct])
```
Comment (oswaldquek) [inline app/controllers/payment-links/get-delete.controller.js:23]: Didn't know we can do this. Nice.

File: app/controllers/payment-links/get-manage.controller.js [hunk 2]
```diff
@@ -10,26 +10,25 @@ const { renderErrorView } = require('../../utils/response')
...
+      productsLength: paymentLinks.length,
+      englishPaymentLinks,
+      welshPaymentLinks,
+      paths,
+      externalServiceId
+    }
+    return response(req, res, 'payment-links/manage', pageData)
+  } catch (err) {
+    logger.error(`[requestId=${req.correlationId}] Get ADHOC product by gateway account id failed - ${err.message}`)
+    return renderErrorView(req, res)
```
Comment (sfount) [inline app/controllers/payment-links/get-manage.controller.js:32]: This PR is great! So much clearer. 

Quick comment that now that we've set up error handling appropriately for the middleware architecture generic error handling can be done downstream of the controller (for custom error handling like redirecting the controller can take of those). 

The following would have the equivalent functionality but would be one less place the developer has to think about responding. 

```javascript
next(err)
```

Comment (kbottla) [reply app/controllers/payment-links/get-manage.controller.js:32]: 👍🏽 Updated in https://github.com/alphagov/pay-selfservice/pull/2471/files#diff-e3c9d7742c1ef0d275b4a361576dda5dd01c427955b6c94663e10f19dc739cd6R29

**Pull request merged**